### ISSUE TITLE
Microsoft EdgeHTML 15 Browser Tools hooking into zone.js

### ISSUE THREAD

**Issue by wizarrc** [angular/zone.js#817]
Originally Fixed in pull request #155 but now with EdgeHTML 15, the browser tools are hooking in at a different callback function.

https://github.com/angular/zone.js/blob/master/lib/common/utils.ts#L368

The latest version of Edge, `data.handler.toString()` does not return `[object FunctionWrapper]` but `function __BROWSERTOOLS_CONSOLE_SAFEFUNC() { [native code] }` instead. I made changes locally on my machine to zone.js/utils.ts to see if the browser events would go away in my performance tab, and sure enough they did. This seems to unnecessarily hook into zone.js, causing a lot of calls when testing performance.

Can anyone confirm this issue is back in EdgeHTML 15? There should be a better way than just check a string, because they have seemed to changed their internal function. Anyways, below is my local changes and works on my machine.

```
var testString = data.handler.toString();
validZoneHandler = data.handler && (testString === '[object FunctionWrapper]' || testString == 'function __BROWSERTOOLS_CONSOLE_SAFEFUNC() { [native code] }');
```

**Update**
Actually this code looks more correct as it handles the null check properly.
```
if(data.handler) {
  var testString = data.handler.toString();
  validZoneHandler = testString === '[object FunctionWrapper]' || testString == 'function __BROWSERTOOLS_CONSOLE_SAFEFUNC() { [native code] }';
}
```

**Comment by JiaLiPassion** [comment]
@wizarrc , thank you for posting the issue, I will check it, I will first try to install edge 15, do you know where to download edge 15? Or I have to take part in the `windows insider program`? Currently my edge's version is `14`

**Comment by wizarrc** [comment]
@JiaLiPassion Microsoft EdgeHTML 15.15063 should be part of the release build in the "Creator's Update". That is the version I am using anyways. I'm not sure if this issue is present in other versions.

**Comment by JiaLiPassion** [comment]
@wizarrc , got it, I will check it.

**Comment by wizarrc** [comment]
Before suggested fix:
<img width="313" alt="before" src="https://user-images.githubusercontent.com/8039397/27208758-8ca1ec80-5215-11e7-998a-08605c1091be.png">
After suggested fix:
<img width="254" alt="after" src="https://user-images.githubusercontent.com/8039397/27208760-8f42342c-5215-11e7-92e7-abc014d8c01b.png">

Notice all the DOM events in the before event. If you follow the link in the javascript code, it refers to a zone.js callback.

**Comment by JiaLiPassion** [comment]
@wizarrc , got it, thank you for the detailed information.

**Comment by wizarrc** [comment]
@JiaLiPassion Thanks for all your work! I can't wait till it gets released so I can test it out.

**Comment by wizarrc** [comment]
This fix looks to be in v0.8.13. I'm closing this.
